paasta_tools/mesos_maintenance.py

```diff
@@ -331,7 +399,110 @@ def load_credentials(mesos_secrets='/nail/etc/mesos-slave-secret'):
                   "See http://mesos.apache.org/documentation/latest/authentication/ for details" % mesos_secrets)
         raise
-    return username, password
+    return Credentials(file=mesos_secrets, principal=username, secret=password)
+
+
+def get_principal(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the principal from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the principal/username
+    """
+    return load_credentials(mesos_secrets).principal
+
+
+def get_secret(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the secret from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the secret/password
+    """
+    return load_credentials(mesos_secrets).secret
+
+
+def reserve(slave_id, resources):
+    """Dynamically reserve resources in marathon to prevent tasks from using them.
+    :param slave_id: the id of the mesos slave
+    :param resources: list of Resource named tuples specifying the name and amount of the resource to (un)reserve
+    :returns: boolean where 0 represents success and 1 is a failure
+    """
+    log.info("Dynamically reserving resoures on %s: %s" % (slave_id, resources))
+    payload = {
+        'slaveId': slave_id,
+        'resources': str(build_reservation_payload(resources)).replace("'", '"').replace('+', '%20')
+    }
+    client_fn = reserve_api()
+    try:
+        reserve_output = client_fn(method="POST", endpoint="", data=payload).text
+    except HTTPError:
+        log.debug("Error adding dynamic reservation.")
+        raise
+    return reserve_output
+
+
+def unreserve(slave_id, resources):
+    """Dynamically unreserve resources in marathon to allow tasks to using them.
+    :param slave_id: the id of the mesos slave
+    :param resources: list of Resource named tuples specifying the name and amount of the resource to (un)reserve
+    :returns: boolean where 0 represents success and 1 is a failure
+    """
+    log.info("Dynamically unreserving resoures on %s: %s" % (slave_id, resources))
+    payload = {
+        'slaveId': slave_id,
+        'resources': str(build_reservation_payload(resources)).replace("'", '"').replace('+', '%20')
+    }
+    client_fn = unreserve_api()
+    try:
+        unreserve_output = client_fn(method="POST", endpoint="", data=payload).text
+    except HTTPError:
+        log.debug("Error adding dynamic unreservation.")
+        raise
+    return unreserve_output
+
+
+def reserve_all_resources(hostnames):
+    mesos_state = get_mesos_master().state_summary()
+    hosts = []
+    components = hostnames_to_components(hostnames)
+    for component in components:
+        hosts.append(component.host)
+    for slave in mesos_state['slaves']:
```

Comment:
how about removing the inner ``if`` here by filtering those that you're interested in.

```
known_slaves = [slave for slave in mesos_state['slaves'] if slave['hostname'] in hostnames]
for slave in known_slaves:
    log.info("Reserving all resources on %s" % hostname))
    ...
```